### PR TITLE
Swap to Underground oak items while mining

### LINKED ISSUE #51: Switch to mining oak items before clearing tiles
### Is your feature request related to a problem? Please describe.
Right now, if I want to level up or gain the bonus from the mining oak items, I need to use up precious slots on items that (barring the battery) provide no buff being active when mining is not happening.

### Describe the solution you'd like
I think the easiest solution would be to, every time a mining operation is about to occur, quickly:
* save the state of current oak items
* equip the Cell Battery, Explosive Charge, and Treasure Scanner (provided they are unlocked)
* Mine the tiles
* return to the saved state of oak items

### Additional context
I am not sure the performance impact this change would have on the script, as it does significantly increase the number of actions taken every mining loop. If it does turn out to have a significant performance impact, I think the interval delay can be increased, ~~as 0.3 seconds is rather often for something that only changes state (without upgrades) once a minute.~~ 

edit as I just realized I misunderstood the code

### PR DESCRIPTION
**Pull request by oohwooh**
I have been unable to fully test this due to how the script is loaded, but the testing I was able to do suggests it will work, unless I am misunderstanding the loops. (if you have any guidance on how to test please let me know, my last PR I was able to inject via console, but as I am modifying existing functions this time I was not able to get that to work)

Before merge, it would be good to add documentation of this feature in the wiki, and a toggle - in addition to further testing.

closes #51

### CODE REVIEW

**Comment by akwawa** [comment]
@oohwooh You can test it by forking this project on your repository and manually changing the variables in the main script:
```
// If my branch is named like "feature-1"
var releaseLabel = "feature-1";

// My repostory on github with my name "Akwawa" :
var pokeclickerAutomationReleaseUrl = "https://raw.githubusercontent.com/Akwawa/pokeclicker-automation/" + releaseLabel + "/";
```

**Comment by akwawa** [comment]
I haven't tested your script but can you adapt the tabulation of the "Underground" variable to match the other variables?

Will it work if you don't have the items?

**Comment by oohwooh** [comment]
Thanks for the suggestion on how to test - not sure how I didn't think of that haha

Just tested and it works as intended.


> Will it work if you don't have the items?

`OakItem.__equipLoadout()` handles this functionality by default

**Comment by Farigh** [comment]
This would only work if no other feature is enabled.
Most feature will update the current loadout as well, overriding the one you chose.
That's why I never bothered equipping any item for mining, as the user could do so by himself.

Your proposition is a good first step, I guess, but its usage would be very limited.

**Comment by oohwooh** [comment]
> Most feature will update the current loadout as well, overriding the one you chose.

What if we added a function to `OakItem` allowing us to "Lock" or "Unlock" the automated changing of oak items. When a loadout tries to be equipped and oak items are locked, the request could either be ignored or added to a queue.

What are your thoughts on this?

File: src/lib/Utils/OakItem.js [hunk 1]
```diff
@@ -39,6 +39,15 @@ class AutomationUtilsOakItem
...
                             ];
+
+        /**
+         * @brief The most efficient setup to mine in the underground
+         */
+        static Underground = [
+                                OakItemType.Cell_Battery,
+                                OakItemType.Explosive_Charge,
+                                OakItemType.Treasure_Scanner,
+        ];
```
Comment (Farigh) [inline src/lib/Utils/OakItem.js:50]:
```suggestion
        static Underground = [
                                 OakItemType.Cell_Battery,
                                 OakItemType.Explosive_Charge,
                                 OakItemType.Treasure_Scanner,
                             ];

```

**Comment by Farigh** [comment]
Not that relevant anymore, since the player can:
  - Disable loadout automation
  - Enable restore item auto-use at any time

**Pull request closed without merging**